File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0

        n = len(nums)

        ans = 0
        s = set(nums)
        for i in range(n):
            if nums[i] - 1 not in s:
                curr_length = 1
                while nums[i] + curr_length in s:
                    curr_length += 1
                ans = max(ans, curr_length)
        return ans

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        var s = new HashSet<Integer>();
        for (int num: nums) {
            s.add(num);
        }

        var ans = 0;
        for (int num: s) {
            if (!s.contains(num - 1)) {
                var curr_length = 0;
                while (s.contains(num + curr_length)) {
                    curr_length++;
                }
                ans = Math.max(ans, curr_length);
            }
        }

        return ans;
    }
}
